src/renderer/hooks/useFlippersMotorCurrents.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useMemo, useState } from 'react';
+import { selectRobotNameState } from '../store/modules/input';
+
+import { store } from '../store/store';
+import { useRosSubscribe } from './useRosSubscribe';
+
+interface FlipperTopic {
+  name: string;
+  messageType: string;
+}
+
+interface FlipperTopics {
+  flipperFLTopic: FlipperTopic;
+  flipperFRTopic: FlipperTopic;
+  flipperRLTopic: FlipperTopic;
+  flipperRRTopic: FlipperTopic;
+}
+
+export const useFlippersMotorCurrents = () => {
+  const robotName = selectRobotNameState(store.getState());
+
+  const [flMotorCurrentState, setFlMotorCurrentState] = useState(0);
+  const [frMotorCurrentState, setFrMotorCurrentState] = useState(0);
+  const [rlMotorCurrentState, setRlMotorCurrentState] = useState(0);
+  const [rrMotorCurrentState, setRrMotorCurrentState] = useState(0);
+  const flipperTopics = useMemo<FlipperTopics>(
+    () => ({
+      flipperFLTopic: {
+        name: `${robotName}/flippers/flipper_fl_motor_current`,
+        messageType: 'std_msgs/Float64',
+      },
+      flipperFRTopic: {
+        name: `${robotName}/flippers/flipper_fr_motor_current`,
+        messageType: 'std_msgs/Float64',
+      },
+      flipperRLTopic: {
+        name: `${robotName}/flippers/flipper_rl_motor_current`,
+        messageType: 'std_msgs/Float64',
+      },
+      flipperRRTopic: {
+        name: `${robotName}/flippers/flipper_rr_motor_current`,
+        messageType: 'std_msgs/Float64',
+      },
+    }),
+    [robotName]
+  );
+
+  // useRosSubscribe for each flipper
```

Comment:
Would there be a way the use a loop instead of duplicating the code? Cycling through topic name and set function.

Reply:
React really doesn't like it when you use hooks conditionally (in if, for loops, etc) So that was the simplest way I found to organize the code for now